### PR TITLE
add process doc for migrating reviewer permissions to owners files

### PR DESCRIPTION
**Pull request by dhellmann**

### CODE REVIEW

**Comment by dhellmann** [comment]
Announced via https://groups.google.com/forum/#!topic/metal3-dev/iVf9sHCJV0g

File: processes/reviewer-permissions-migration.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+reviewers to the owners files. When we have merged those PRs, we can
+change Prow’s configuration to have it use the OWNERS file instead of
+github org membership for /lgtm permissions. We should also update the
+maintainers process document to include instructions for managing the
+list of org members and for managing the reviewer list for a repo.
+
+### Risks and Mitigations
+
+If we miss a repository, the list of approvers for that repository
+will also have reviewer permission so we can still merge patches.
```
Comment (russellb) [inline processes/reviewer-permissions-migration.md:76]: and if all else fails, the github org owners can manually push changes

**Comment by russellb** [comment]
In general I'm +0 on this.  It's more fine grained control (good), but more overhead (not so good).

Since merging code requires `approved` + `lgtm` and we already carefully managed `approved` access, I don't have a problem being very generous with `lgtm` access.

If we go this route, I can handle the prow config side when the time comes.  I havn't found a config example, but I can see it in this part of the code where it handles `skipCollaborators`: https://github.com/kubernetes/test-infra/blob/master/prow/plugins/lgtm/lgtm.go#L265-L307

**Comment by maelk** [comment]
looks good to me! This would give more fine-tuning possibilities if needed, per repo or even more fine-grained if needed. Maybe we could already draft a process for adding reviewers, so that we engage into this change with a picture of the end goal.

**Comment by russellb** [comment]
> looks good to me! This would give more fine-tuning possibilities if needed, per repo or even more fine-grained if needed. Maybe we could already draft a process for adding reviewers, so that we engage into this change with a picture of the end goal.

For a process, how about just sign-off from 2 existing approvers?  That's already to get the PR merged, so just make that the procedure?  It should be a relatively low bar to get added to reviewers.

**Comment by maelk** [comment]
agreed, 2 approvers signing on the PR to add the reviewer sounds good to me!

**Comment by dhellmann** [comment]
> agreed, 2 approvers signing on the PR to add the reviewer sounds good to me!

See #94

**Comment by dhellmann** [comment]
I propose that we use lazy consensus rules to approve this by 12 June. That's a couple of days after our next community meeting, which gives us plenty of time to discuss it here as well as synchronously.

**Comment by dhellmann** [comment]
Today I created PRs for all repositories where the reviewer list would have new members based on the policy described here and the command

```
hub api repos/metal3-io/$(basename $(pwd))/stats/contributors | jq -r '.[] | if .total >=10 then " - " + .author.login else "" end'
```

**Comment by dhellmann** [comment]
> Today I created PRs for all repositories where the reviewer list would have new members based on the policy described here and the command
> 
> ```
> hub api repos/metal3-io/$(basename $(pwd))/stats/contributors | jq -r '.[] | if .total >=10 then " - " + .author.login else "" end'
> ```

At least one PR is reporting an error because OWNERS mentions someone who is not a member of the org. So I think we're going to have to change the Prow settings, then invite people, then add them to OWNERS.

File: processes/reviewer-permissions-migration.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+so we're going to want to have a separate process for that.
+
+If we focus on the transition for now, we can define that process
+separately later. So, I propose that we take the contributors with 10
+or more commits according to github’s contribution list (via the page
+like
+<https://github.com/metal3-io/cluster-api-provider-metal3/graphs/contributors>)
+as the initial set of reviewers for each repo. That will allow us to
+complete the migration and we can expand the list further afterwards.
+
```
Comment (jan-est) [inline processes/reviewer-permissions-migration.md:65]: This is a good starting point to get enough reviewers to each repo.

File: processes/reviewer-permissions-migration.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+1. Transition from github org membership to OWNERS files for reviewer
+   permissions.
+
+### Non-Goals
+
+1. Define a new process for approving reviewers.
+2. Change the process for approving approvers.
+3. Change the permissions for managing the CI infrastructure.
+
```
Comment (jan-est) [inline processes/reviewer-permissions-migration.md:40]: I am +1 with these non-goals as long as a process of adding the new reviewers is quick. I am only bit concern how we will give reviewers rights, /approve and /lgtm rights in the future. I totally agree that our growing projects needs some sort of gatekeeping, but how and when should we define the process for approving reviewers and promotion of members to have /lgtm for example?

Comment (maelk) [reply processes/reviewer-permissions-migration.md:40]: this was defined in #94

Comment (jan-est) [reply processes/reviewer-permissions-migration.md:40]: Thanks @maelk everything was well defined on #94

**Pull request merged**